sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -2012,4 +2012,12 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase {
     new SparkException(
       errorClass = "MULTI_VALUE_SUBQUERY_ERROR", messageParameters = Array(plan), cause = null)
   }
+
+  def comparisonResultIsNullError(): Throwable = {
+    new NullPointerException(
```

Comment:
I remember the exception being thrown should inherit `SparkThrowable` when it's thrown to end-users ... cc @MaxGekk am I remembering it correct?

Reply:
yea, with the new error class framework, the java exception type doesn't matter any more. We can always use `SparkException`, but we need to add an error class for it.